Update login logic for admin (Fixes #3736)

adithyanotfound opened the pull request:
**What kind of change does this PR introduce?**

Bugfix

**Issue Number:**

Fixes #3736 

**Snapshots/Videos:**

https://github.com/user-attachments/assets/7c1b1be4-9e8c-4126-9daf-b3b183a5372a

**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**

Admins will now be redirected to the admin dashboard, regardless of whether they log in via the user login page or the admin login page.

**Does this PR introduce a breaking change?**

No

## Checklist

### CodeRabbit AI Review
- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [x] I have implemented or provided justification for each non-critical suggestion
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Bug Fixes**
	- Improved accessibility on the login page by adding alt attributes to social media icons.
	- Resolved warnings by removing unused parameters and commented-out code.
	- Updated navigation logic to better handle user and admin roles during login.

- **Tests**
	- Enhanced login page tests with new user login and error handling scenarios.
	- Improved test structure and coverage for registration and organization selection.

- **Chores**
	- Updated ESLint configuration for better TypeScript support.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

adithyanotfound commented on src/screens/LoginPage/LoginPage.spec.tsx:
```diff
@@ -1104,48 +1106,61 @@ describe('Testing redirect if already logged in', () => {
-      origin: 'https://localhost:4321',
-      pathname: '/',
-    },
+  it('Redirects based on role (admin or user)', async () => {
+    const isAdmin = true;
+    const mockNavigate = vi.fn();
+    const navigate = (url: string) => mockNavigate(url);
+
+    if (isAdmin) {
+      navigate('/orglist');
+    } else {
+      navigate('/user/organizations');
+    }
+
+    expect(mockNavigate).toHaveBeenCalledWith(
+      isAdmin ? '/orglist' : '/user/organizations',
+    );
   });
```
Not required

adithyanotfound commented on src/screens/LoginPage/LoginPage.spec.tsx:
```diff
@@ -26,7 +26,9 @@
 import { BACKEND_URL } from 'Constant/constant';
 import useLocalStorage from 'utils/useLocalstorage';
-import { vi, beforeEach, expect, it, describe } from 'vitest';
+import { vi, beforeEach, expect, it, describe, Mock } from 'vitest';
+import { toast } from 'react-toastify';
+import { error } from 'console';
```
Done

Cioppolo14 commented:
@adithyanotfound Please fix the code coverage.

palisadoes commented:
@adithyanotfound 

Is this the approach being used?

- https://github.com/PalisadoesFoundation/talawa-admin/issues/3736#issuecomment-2817007686

adithyanotfound commented:
> Is this the approach being used?

@palisadoes No. With the updated logic admin cannot log in as a user.

adithyanotfound commented:
> 1. Please get the patch closer to 100% coverage
> 2. Why isn't the CodeRabbit test passing? It used to work and I remember you fixed it in the past.

@palisadoes 

1. I'll start working on the tests. Please confirm if the admin login logic is correct. Previously, admins had to navigate to `/admin` to log in. In this PR, I’ve updated it so that admins can now log in through the regular user login screen as well. However, with this change, admins can no longer log in as regular users. @bandhan-majumder had brought up this issue earlier.

2. Please revert PR https://github.com/PalisadoesFoundation/talawa-admin/pull/3942.
It removed index.html from sensitive files and updated the coderabbitai check.

bandhan-majumder commented:
Yes @adithyanotfound . We need to decide whether to have this feature or not. If yes, then why do we even need `/admin` route to login for. A single login screen would be perfect to determine the role and show relevant screens. Having another admin specific login screen would be repetitive. However, admins can never access the screens of regular users which might not be what we want. 

WDYT @palisadoes

adithyanotfound commented on .github/workflows/pull-request-review.yml:
```diff
@@ -3,13 +3,14 @@ name: Pull Request Review
   pull_request_review:
     types: [submitted]
-
+    
```
will fix

adithyanotfound commented on src/screens/LoginPage/LoginPage.spec.tsx:
```diff
@@ -1104,48 +1138,98 @@ describe('Testing redirect if already logged in', () => {
+    expect(mockNavigate).toHaveBeenCalledWith('/user/organizations');
+  });
+
+  it('Returns toast warning with wrong credentials', async () => {
+    render(
+      <MockedProvider addTypename={false} link={userLink}>
+        <BrowserRouter>
+          <Provider store={store}>
+            <I18nextProvider i18n={i18nForTest}>
+              <LoginPage />
+            </I18nextProvider>
+          </Provider>
+        </BrowserRouter>
+      </MockedProvider>,
+    );
+
+    await wait(); // wait for any initial page loads
+    await userEvent.type(
+      screen.getByTestId(/loginEmail/i),
+      'wrong-user@example.com',
+    );
+    await userEvent.type(
+      screen.getByPlaceholderText(/Enter Password/i),
+      'wrong-password',
+    );
+    await userEvent.click(screen.getByTestId('loginBtn'));
+    await wait(); // wait for toast/error after login click
+
+    expect(toast.warn).toHaveBeenCalledWith('Not found');
+  });
```
Not required since we need to provide wrong creds

adithyanotfound commented on .github/workflows/pull-request-review.yml:
```diff
@@ -20,25 +22,31 @@ jobs:
             // Filter reviews that have a user login containing "coderabbit" (case-insensitive)
             const codeRabbitReviews = reviews.filter(review => 
-              review.user.login.toLowerCase().includes('coderabbit')
+              review.user && review.user.login && review.user.login.toLowerCase().includes('coderabbit')
```
will do

adithyanotfound commented on eslint.config.js:
```diff
@@ -41,6 +41,7 @@ export default [
         ecmaFeatures: {
           jsx: true,
         },
+        lib: ['dom', 'esnext'],
       },
```
not required

palisadoes commented:
Closing as abandoned
